### PR TITLE
feat: add machines endpoint

### PR DESCRIPTION
**Pull request by rkr090**

### CODE REVIEW

File: lib/my_john_deere_api/model/machine_engine_hour.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    private
+
+    def map_attributes(record)
+      @reading = record['reading']
+      @report_time = record['reportTime']
+    end
+
+    def expected_record_type
+      'EngineHours'
```
Comment (solsberg) [inline lib/my_john_deere_api/model/machine_engine_hour.rb:13]: the record type is plural but the model name is singular, which is correct?

File: lib/my_john_deere_api/validators/machine_breadcrumb.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    ##
+    # Custom validations for this class
+
+    def validate_attributes
+      # validate_speed_data
+      # validate_heading_data
+      # validate_machine_state_data
+      # validate_fuel_level_data
+      # validate_point_data
```
Comment (solsberg) [inline lib/my_john_deere_api/validators/machine_breadcrumb.rb]: why are these commented out?

File: lib/my_john_deere_api/validators/machine_device_state_report.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    def required_attributes
+      [:machine_id, :time, :location, :gps_fix_timestamp]
+    end
+
+    ##
+    # Custom validations for this class
+
+    def validate_attributes
+      # validate_location_data
```
Comment (solsberg) [inline lib/my_john_deere_api/validators/machine_device_state_report.rb]: commented out?

**Review by solsberg — APPROVED** [review]
👍 looks good, just a couple of comments

File: lib/my_john_deere_api/model/machine_engine_hour.rb (hunk 1 above)
Comment (rkr090) [reply lib/my_john_deere_api/model/machine_engine_hour.rb:13]: Yes, the API is returning the type as "EngineHours"

File: lib/my_john_deere_api/validators/machine_breadcrumb.rb (hunk 2 above)
Comment (rkr090) [reply lib/my_john_deere_api/validators/machine_breadcrumb.rb]: Fixed

File: lib/my_john_deere_api/validators/machine_device_state_report.rb (hunk 3 above)
Comment (rkr090) [reply lib/my_john_deere_api/validators/machine_device_state_report.rb]: Fixed

**Review by bknarendra — APPROVED** [review]

**Pull request merged**